src/mesh/mesh.cxx

```diff
@@ -738,3 +760,59 @@ constexpr decltype(MeshFactory::type_name) MeshFactory::type_name;
 constexpr decltype(MeshFactory::section_name) MeshFactory::section_name;
 constexpr decltype(MeshFactory::option_name) MeshFactory::option_name;
 constexpr decltype(MeshFactory::default_type) MeshFactory::default_type;
+
+int Mesh::getCommonRegion(int lhs, int rhs) {
+  if (lhs == rhs) {
+    return lhs;
+  }
+  const int low = std::min(lhs, rhs);
+  const int high = std::max(lhs, rhs);
+  if (low == -1) {
+    return high;
+  }
+  /* Memory layout of indices
+   * left is lower index, bottom is higher index
+   *    0  1  2  3
+   * 0
+   * 1  0
+   * 2  1  2
+   * 3  3  4  5
+   * 4  6  7  8  9
+   *
+   * As we only need half of the square, the indices do not depend on
+   * the total number of elements.
+   */
+  const size_t pos = (high * (high - 1)) / 2 + low;
+  if (region3Dintersect.size() <= pos) {
+    BOUT_OMP(critical(mesh_intersection_realloc))
+#if BOUT_USE_OPENMP
+    if (region3Dintersect.size() <= pos)
+#endif
```

Comment:
Is there another way to write this without the `#ifdef` guards and repeated conditionals? Maybe `BOUT_OMP(single)`? If not, then I think this needs a good comment to ensure it doesn't get accidentally removed

Reply:
I added comments. Please let me know if there is still anything unclear.